src/components/Gradebook/SearchControls.test.jsx

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+
+import { SearchControls } from './SearchControls';
+
+describe('SearchControls', () => {
+  let props;
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+    props = {
+      courseId: 'course-v1:edX+DEV101+T1',
+      filterValue: 'alice',
+      setGradebookState: jest.fn().mockName('setGradebookState'),
+      showSpinner: false,
+      toggleFilterDrawer: jest.fn().mockName('toggleFilterDrawer'),
+      // From Redux
+      getUserGrades: jest.fn().mockName('getUserGrades'),
+      searchForUser: jest.fn().mockName('searchForUser'),
+      selectedAssignmentType: 'homework',
+      selectedCohort: 'spring term',
+      selectedTrack: 'masters',
+    };
+  });
+
+  const searchControls = (overriddenProps) => {
+    props = Object.assign(props, overriddenProps);
+    return shallow(<SearchControls {...props} />);
+  };
+
+  describe('Component', () => {
+    describe('spinner', () => {
+      // These tests aren't strictly necessary since they're already covered by snapshot tests
+      it('shows a spinner overlay when the page is waiting for some action to complete', () => {
+        const wrapper = searchControls({ showSpinner: true });
+        expect(wrapper.find('.spinner-overlay').exists()).toEqual(true);
+      });
+
+      it('does not show a spinner overlay when page indicates it is not loading', () => {
+        const wrapper = searchControls();
+        expect(wrapper.find('.spinner-overlay').exists()).toEqual(false);
+      });
+    });
+
+    describe('onSubmit', () => {
+      it('calls props.searchForUser with correct data', () => {
+        const wrapper = searchControls();
+        wrapper.instance().onSubmit('bob');
+
+        expect(props.searchForUser).toHaveBeenCalledWith('course-v1:edX+DEV101+T1', 'bob', 'spring term', 'masters', 'homework');
```

Comment:
toHaveBeenCalledWith args should be split out to their own lines.

Reply:
*same comment for other places with multi-arg call checks below.  don't want to spam

Reply:
So I basically limit this styling to what ESLint complains about (since I have it integrated with my IDE). Valid nit but still a nit.

Reply:
Would prefer we have an enforced, auto-runnable style system so we don't have to have to spend time with this but that's a separate issue for the frontend people to figure out 🙃